pkg/v22/resource/app/desired.go

```diff
@@ -200,11 +207,20 @@ func (r *Resource) newAppSpecs() []key.AppSpec {
 	}
 }
 
-func newUserConfig(clusterConfig v1alpha1.ClusterGuestConfig, appSpec key.AppSpec, configMaps map[string]corev1.ConfigMap, secrets map[string]corev1.Secret) g8sv1alpha1.AppSpecUserConfig {
+func newUserConfig(clusterConfig v1alpha1.ClusterGuestConfig, appSpec key.AppSpec, configMaps, tenantConfigMaps map[string]corev1.ConfigMap, secrets map[string]corev1.Secret) (g8sv1alpha1.AppSpecUserConfig, error) {
 	userConfig := g8sv1alpha1.AppSpecUserConfig{}
 
-	_, ok := configMaps[key.AppUserConfigMapName(appSpec)]
-	if ok {
+	_, cmExists := configMaps[key.AppUserConfigMapName(appSpec)]
+	tenantCM, tenantExists := tenantConfigMaps[key.AppUserConfigMapName(appSpec)]
+
+	// A tenant configmap exists with user settings but the user configmap for
+	// this app CR does not exist yet. We delay creating the app CR until it
+	// does so the app is installed with the correct settings.
+	if tenantExists && len(tenantCM.Data) > 0 && !cmExists {
```

Comment:
We create the user values configmaps automatically. We only need to delay if keys have been added.